### PR TITLE
updated homepage with link to challenges, updated challenges ui

### PR DESCRIPTION
**Pull request by deej-tsn**

### CODE REVIEW

File: src/components/CodeChallengeCard.astro [hunk 1]
```diff
@@ -0,0 +1,61 @@
+---
+interface Props {
+    title : string,
+    href : string,
+    difficulty : string
+}
+
+const {href, title, difficulty} = Astro.props;
```
Comment (webtechalex) [inline src/components/CodeChallengeCard.astro:8]: You declared a `Props` interface but I don't think you used it. Should the interface annotate this definition?

Comment (deej-tsn) [reply src/components/CodeChallengeCard.astro:8]: As i understand it, the props interface tells astro that i will give the code challenge component these parameters, and then in the layout part, i am using this parameters

Comment (deej-tsn) [reply src/components/CodeChallengeCard.astro:8]: Screenshot 2024-10-23 at 16 39 35

used here

Comment (webtechalex) [reply src/components/CodeChallengeCard.astro:8]: That part is fine. But where is the type safety if you declare an interface without using it?

Comment (webtechalex) [reply src/components/CodeChallengeCard.astro:8]: In my mind, line 8 should be:
```
const {href, title, difficulty}: Props = Astro.props;
```
It looks like we have the same problem in the Layout file, but I only just noticed it

Comment (webtechalex) [reply src/components/CodeChallengeCard.astro:8]: Although in my editor, when I look at the Layout props, it somehow knows that `title` is a `string`.
Will resolve this comment for now as it's my understanding that's at fault, I think

**Review by webtechalex — APPROVED** [review]

**Pull request merged**